### ISSUE TITLE
Penalisation of bias and BatchNorm parameters

### ISSUE THREAD

**Issue by Mamiglia** [pytorch/pytorch#115935]
### 📚 The doc issue

Currently, the most common example given when introducing optimisers[[1]](https://pytorch.org/tutorials/beginner/basics/optimization_tutorial.html) is something like:
```Python
optimizer = torch.optim.SGD(model.parameters(), lr=learning_rate)
```
This is of course a good first example, but perhaps we should consider that many users don't realise that `.parameters()` also contains the biases of the layers and the BatchNorm parameters, as it can be seen in #2639 , #1402 ,  [[2]](https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994/16), [[3]](https://discuss.pytorch.org/t/weight-decay-only-for-weights-of-nn-linear-and-nn-conv/114348). 

From what I understand of the current state of research [[4]](https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994/16), it's unclear whether penalising these parameters is a good practice or not, since these parameters should not induce overfitting.

I think it's a problem that this issue is not addressed anywhere in the documentation, as it should be clear that the default behaviour and examples adopt a practice that is not validated by current research.

### Suggest a potential alternative/fix

I believe that this issue should be addressed somewhere, possibly in the [torch.optim](https://pytorch.org/docs/stable/optim.html#module-torch.optim) page, maybe by specifying it [here](https://pytorch.org/docs/stable/optim.html#constructing-it).

In the same page we could change the provided example for Per-Parameter options to one in which the weight decay is used, as this is the current recommended way in which one should address the separation of parameter penalties. _Could be_ something like:
```python
penalized     = [p for name,p in self.named_parameters() if 'bias' not in name]
not_penalized = [p for name,p in self.named_parameters() if 'bias' in name]
optimizer_params = [
    {'params': penalized, 'weight_decay':1e-3,  'lr'=1e-3},
    {'params': not_penalized, 'weight_decay':0}
]
optim.SGD(optimizer_params, lr=1e-2, momentum=0.9)
```

Also I believe it may be a nice idea to include this in the [optimization tutorial](https://pytorch.org/tutorials/beginner/basics/optimization_tutorial.html) somewhere near the end.

-------
This is my first contribution here on pytorch, I hope that this issue is understable, respects the contributing guidelines and it's relevant. In case this change would be welcome I would be glad to implement this myself.

cc @svekars @brycebortree @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

**Comment by janeyx99** [comment]
Hi @Mamiglia! Sorry it's taken me so long to see this and thank you for raising this to our attention. At first glance, I do agree with documenting this discrepancy somewhere, and the place in the docs you linked regarding param groups seems the right place. I have added this as a discussion topic in our next group meeting to verify with the team that this is something we'd like to do and I'll update this thread once that discussion happens (should be within the next 2 weeks).

**Comment by janeyx99** [comment]
As the current research status of this topic is unclear, and since PyTorch is mainly a framework offering tools for doing math, I don't think it is our call to state one way or the other. However, we agree that the current docs can be improved regarding how one could do this. I did notice that keras has a particular "exclude_from_weight_decay" list https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam#exclude_from_weight_decay for this purpose without mentioning the discrepancy. Thus, we would accept PRs for the following:

1. Adding an example like the one you mentioned into our torch docs where you recommended, potentially after the per-layer example.
2. Mentioning in the tutorial that we have a param groups API with which this can be done (also with an example).
